### ISSUE TITLE
`--nodeep_execroot` doesn't apply

### ISSUE THREAD

**Issue by keith** [bazelbuild/bazel#12271]
[The help](https://docs.bazel.build/versions/master/command-line-reference.html#flag--deep_execroot) for `--deep_execroot` (on by default) says:

> If set, the execution root will be under $OUTPUT_BASE/execroot instead of $OUTPUT_BASE.

But using this flag doesn't seem to change this value (for `info` or `build`):

```
% bazel info execution_root
Starting local Bazel server and connecting to it...
/private/var/tmp/_bazel_ksmiley/ce67f9c12271421120de449ea6206e4e/execroot/build_bazel_rules_apple

% bazel --nodeep_execroot info execution_root
WARNING: Running Bazel server needs to be killed, because the startup options are different.
Starting local Bazel server and connecting to it...
/private/var/tmp/_bazel_ksmiley/ce67f9c12271421120de449ea6206e4e/execroot/build_bazel_rules_apple
```

### What operating system are you running Bazel on?

macOS

### What's the output of `bazel info release`?

release 3.6.0

**Comment by aiuto** [comment]
It looks like the code does not use it. A search through log history points to a gradual removal of the flag.

4873e56c6ce2c29650f2197eb492179b341e872a
   Get rid of --nodeep_execroot reference
   Not that it does anything, but let's not encourage people.

af4c3a9bb9bfd3d56ab56dcc4f14a9318e5ec4b5
  Flip --deep_execroot to default to true

 It looks like we were migrating to the new new behavior, removed all the references to the flag, and then forgot to remove the flag.

Are you expecting it to do anything?
Would you be willing to change this issue to  "remove startup option deep_execroot", because that is the reasonable thing to do.

**Comment by keith** [comment]
I was hoping it would remove the `execroot/` from the path to save some characters in absolute paths that are causing us to exceed ARG_MAX. But if the plan is to remove this knob that's fine. Here's the change to remove it https://github.com/bazelbuild/bazel/pull/12282

**Comment by aiuto** [comment]
Closed by #12282
